### PR TITLE
Instances for MonadBaseControl and MonadTransControl

### PR DESCRIPTION
**Pull request by Lupino**
On my project I need HaskeyT instance MonadBaseControl

### CODE REVIEW

File: haskey-mtl.cabal [hunk 1]
```diff
@@ -29,7 +29,9 @@ library
     mtl                     >=2.1  && <3,
     transformers            >=0.3  && <1,
     haskey-btree            >=0.3 && <0.4,
-    haskey                  >=0.3 && <0.4
+    haskey                  >=0.3 && <0.4,
+    monad-control           >=1.0.2.0 && <1.0.2.4,
```
Comment (hverr) [inline haskey-mtl.cabal]: Could explain the lower bound on `monad-control`? Would `1.0.1.0` work as a lower bound? (To support LTS 6)

File: src/Control/Monad/Haskey.hs [hunk 2]
```diff
@@ -42,6 +44,8 @@ import qualified Control.Monad.State.Lazy as StateL
 import qualified Control.Monad.State.Strict as StateS
 import qualified Control.Monad.Writer.Lazy as WriterL
 import qualified Control.Monad.Writer.Strict as WriterS
+import Control.Monad.Trans.Control
+import Control.Monad.Base
```
Comment (hverr) [inline src/Control/Monad/Haskey.hs]: Could you sort these imports and only import explicitly what you need, using the `import Module (...)` syntax

**Comment by Lupino** [comment]
@hverr thinks for advice， I change the code right now

**Comment by hverr** [comment]
One last small thing: Lupino/haskey-mtl#1

**Comment by hverr** [comment]
Available in [haskey-mtl-0.3.1.0](https://hackage.haskell.org/package/haskey-mtl-0.3.1.0)

**Pull request merged**